### PR TITLE
Add deprecation warning for JavaPluginConvention

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test RFM

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/NaggingJavaPluginConvention.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    @Override
+    public boolean getAutoTargetJvmDisabled() {
+        logDeprecation();
+        return delegate.getAutoTargetJvmDisabled();
+    }
+
+    private static void logDeprecation() {
+        DeprecationLogger.deprecateType(DefaultJavaPluginConvention.class)
+            .willBeRemovedInGradle9()
+            .withUpgradeGuideSection(7, "java_convention_deprecation")
```
Comment (jbartok) [inline subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/NaggingJavaPluginConvention.java:179]: Note to self: wrong upgrade guide, don't forget to fix this!

**Review by jvandort — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Comment by TWiStErRob** [comment]
FYI: https://youtrack.jetbrains.com/issue/IDEA-320266

**Pull request merged**